File: frontend/src/types/organization.ts
export interface OrgAdmin {
  id: string;
  name: string;
  bc_company_name: string;
  bc_database_url: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface OrgCreatePayload {
  name: string;
  bc_company_name: string;
  bc_database_url?: string | null;
}

export interface OrgUpdatePayload {
  name?: string;
  bc_company_name?: string;
  bc_database_url?: string | null;
  is_active?: boolean;
}

File: backend/src/models/organization.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from src.core.database import Base


class Organization(Base):
    __tablename__ = "organizations"

    id: Mapped[str] = mapped_column(
        String(36), primary_key=True, default=lambda: str(uuid.uuid4())
    )
    name: Mapped[str] = mapped_column(String(200), nullable=False)
    # BC SQL Server table prefix, e.g. "BYS" → tables like [BYS$Item]
    bc_company_name: Mapped[str] = mapped_column(String(100), nullable=False)
    # Optional per-org BC ODBC connection string; NULL = use BC_DATABASE_URL env var
    bc_database_url: Mapped[str | None] = mapped_column(Text, nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), onupdate=func.now(), nullable=False
    )

    users: Mapped[list["User"]] = relationship(back_populates="organization")  # noqa: F821
